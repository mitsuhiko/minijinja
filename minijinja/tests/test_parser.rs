#![cfg(feature = "unstable_machinery")]
use minijinja::machinery::parse;

#[test]
fn test_parser() {
    insta::glob!("parser-inputs/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let filename = path.file_name().unwrap().to_str().unwrap();
        let ast = parse(&contents, filename, Default::default(), Default::default());
        insta::with_settings!({
            description => contents.trim_end(),
            omit_expression => true,
        }, {
            insta::assert_debug_snapshot!(&ast);
        });
    });
}
