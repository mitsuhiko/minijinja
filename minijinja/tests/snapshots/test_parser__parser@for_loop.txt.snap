---
source: minijinja/tests/test_parser.rs
expression: "&ast"
input_file: minijinja/tests/parser-inputs/for_loop.txt

---
Ok(
    Template {
        children: [
            EmitRaw {
                raw: "<ul>\n",
            } @ 1:0-2:0,
            ForLoop {
                target: Var {
                    id: "item",
                } @ 2:7-2:11,
                iter: Var {
                    id: "seq",
                } @ 2:15-2:18,
                filter_expr: None,
                body: [
                    EmitRaw {
                        raw: "\n    <li>",
                    } @ 2:21-3:8,
                    EmitExpr {
                        expr: Var {
                            id: "item",
                        } @ 3:11-3:15,
                    } @ 3:8-3:18,
                    EmitRaw {
                        raw: "</li>\n",
                    } @ 3:18-4:0,
                ],
                else_body: [],
            } @ 2:3-4:12,
            EmitRaw {
                raw: "\n</ul>",
            } @ 4:12-5:5,
        ],
    } @ 1:0-5:5,
)
